mautic/mautic
# EntityResultHelper not needed in API controller dependencies

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [enhancement]
| Deprecations?                          | [n]
| BC breaks? (use the c.x branch)        | [n]
| Automated tests included?              | [y, the API is well-tested] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

I noticed that the `EntityResultHelper` service is required for all API controllers. However, it's just a class with no dependencies. It doesn't have to be a service. So by creating the `EntityResultHelper` when we need it instead of passing it by a controller simplifies all the API controllers.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. You can make some API calls manually to test this, but we have all the API endpoints covered by the API Library tests that I'll execute on this PR. So no manual tests are needed. PHPSTAN would also found any issue with controller dependencies now that we use ID.
